Library/Homebrew/cmd/reinstall.rb

```diff
@@ -8,7 +8,10 @@ module Homebrew
   def reinstall
     FormulaInstaller.prevent_build_flags unless DevelopmentTools.installed?
 
-    ARGV.resolved_formulae.each { |f| reinstall_formula(f) }
+    ARGV.resolved_formulae.each do |f|
+      raise "#{f.full_name} is pinned. You must unpin it to reinstall." if f.pinned?
```

Comment:
I wonder if it's worth making this an `onoe` and using `next` so it'll print a warning for all of them rather than just the first?